butterfly-slack/src/main/java/com/paypal/butterfly/slack/SlackNotifierConfig.java

```diff
@@ -0,0 +1,121 @@
+import com.paypal.butterfly.api.TransformationResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Scope;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+import org.json.JSONObject;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.io.File;
+
+/**
+ * TODO
+ */
+@Configuration
+public class SlackNotifierConfig {
+    private static final Logger logger = LoggerFactory.getLogger(SlackNotifierConfig.class);
+    @Bean
+    @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+    public TransformationListener slackNotification(){
+        return new TransformationListener() {
+            @Override
+            public void postTransformation(TransformationRequest transformationRequest, TransformationResult transformationResult) {
+                postNotification(transformationResult, "postSuccess.json");
+            }
+
+            @Override
+            public void postTransformationAbort(TransformationRequest transformationRequest, TransformationResult transformationResult) {
+                postNotification(transformationResult, "postAbort.json");
+            }
+
+            private void postNotification(TransformationResult transformationResult, String jsonFile){
+                String jsonPath = new File("").getAbsolutePath().concat("/butterfly-slack/src/main/resources/" + jsonFile);
+                String token = System.getProperty("slack.token");
+                try {
+                    String jsonFileString = new String(Files.readAllBytes(Paths.get(jsonPath)));
+                    Client client = ClientBuilder.newClient();
+                    WebTarget target = client.target("https://www.slack.com/api/");
+                    String[] butterflyUserIDs = transformationResult.getUserId().split(" ");
+                    butterflyUserIDs = new String[]{"edwardhwang@berkeley.edu", "ehwang0824@gmail.com"};
+                    String requestID = transformationResult.getId();
+                    for (String id : butterflyUserIDs) {
+                        //TODO: Change hard coded email into butterflyUserIDs
+                        String email = id + ""; //id + "@paypal.com";
+                        String slackUserID = getSlackUserID(target, token, email);
+                        String slackChannelID = getSlackChannelID(target, token, slackUserID);
+                        postSlackMessage(target, token, slackChannelID, jsonFileString, requestID);
+                    }
+                } catch (IOException e) {
+                    logger.warn("No JSON message found for slack notification");
```

Comment:
Add the exception as the second parameter to this log call. If you don't do it, the exception will be swallowed.